boards/nucleo-f746zg/include/periph_conf.h

```diff
@@ -72,23 +131,129 @@ static const uart_conf_t uart_config[] = {
         .tx_af      = GPIO_AF7,
         .bus        = APB1,
         .irqn       = USART2_IRQn,
-#ifdef UART_USE_DMA
-        .dma_stream = 4,
+#ifdef MODULE_PERIPH_DMA
+        .dma        = 2,
         .dma_chan   = 4
 #endif
     }
 };
 
 #define UART_0_ISR          (isr_usart3)
-#define UART_0_DMA_ISR      (isr_dma1_stream6)
 #define UART_1_ISR          (isr_usart6)
-#define UART_1_DMA_ISR      (isr_dma1_stream5)
 #define UART_2_ISR          (isr_usart2)
-#define UART_2_DMA_ISR      (isr_dma1_stream4)
 
 #define UART_NUMOF          ARRAY_SIZE(uart_config)
 /** @} */
 
+/**
+ * @name   SPI configuration
+ *
+ * @{
+ */
+static const uint8_t spi_divtable[2][5] = {
+    {       /* for APB1 @ 54000000Hz */
+        6,  /* -> 421875Hz */
+        5,  /* -> 843750Hz */
+        4,  /* -> 1687500Hz */
+        2,  /* -> 6750000Hz */
+        1   /* -> 13500000Hz */
+    },
+    {       /* for APB2 @ 108000000Hz */
+        7,  /* -> 421875Hz */
+        6,  /* -> 843750Hz */
+        5,  /* -> 1687500Hz */
+        3,  /* -> 6750000Hz */
+        2   /* -> 13500000Hz */
+    }
+};
+
```

Comment:
You can remove the divtable declaration and use the common define in https://github.com/RIOT-OS/RIOT/blob/master/boards/common/stm32/include/cfg_spi_divtable.h because it already defines the same values.

Just add `#include "cfg_spi_divtable.h"` under https://github.com/RIOT-OS/RIOT/pull/14731/files#diff-c93eeec96927441066fd1246376c06e4R23.

```suggestion
```

Reply:
Yes but no, that is questionable : cfg_spi_divtable.h defines 210, 421, 843, 3375 and 6750KHz for APB1 and twice 421, 843, 3375 an 6750KHz for APB2 which is inconsistent for the lowest speed and it is a shame to be limited to 6,75MHz on a 50MHz capable device as I already explained in my previous comment.

So I understand your request on the principle for the homogeneity of the system, backward compatibility, bla bla... although I think that the auto-generation algorithm of spi_divtable is stupid. Sorry.

Even if they had the same values, they are so far different from this MCU possibilities that I think it is better to define them in periph_conf.h to ease their customization.

In our project we previously used a much less powerful atmega328p with an 8MHz SPI clock speed so you can hear that the auto-generated spi_divtable isn't usable in the real world.

Finally, be free to do it yourself if you really want, I am free to carry on using EXTERNAL BOARD_DIRS, that's the game !

Reply:
Hello @hugueslarrive.

[I've fiddled around a bit](https://github.com/bergzand/RIOT/tree/pr/stm32/dynamic_spi_freqs) with the stm32 SPI implementation to get support for arbitrary SPI clock speeds. This completely removes the need for the `spi_divtable` and allows for supplying arbitrary clock speeds as `spi_clock_t` param. Would this solve your issue with the current SPI implementation on the stm32 platform?

Reply:
With this patch and tests/periph_spi I get:
- init 0 0 0 3 4: 1687500Hz
- init 0 0 1 3 4: 54000000Hz
- init 0 0 2 3 4: 843750Hz
- init 0 0 3 3 4: 3375000Hz
- init 0 0 4 3 4: 6750000Hz

Couldn't we just extend spi_divtable to 8 clock speeds, so we could use all the possible speeds on the STM32s without having to make a choice in periph_conf.h?